src/__tests__/utils.spec.ts

```diff
@@ -1,270 +1,184 @@
-        mockVisibilitySelectionMap
-      );
-      expect(payload).toEqual({
-        agent1: ['state1'],
-        agent2: ['agent2'],
-        agent3: [],
-        agent4: ['state1', 'state2'],
-        agent5: ['state1'],
+  });
+
+  describe('getSelectionAfterChildCheckboxClick', () => {
+    it('removes child from parents selection array if it was previously included', () => {
+      const result = getSelectionAfterChildCheckboxClick('child', 'parent', {
+        parent: ['child'],
       });
+      expect(result).toEqual({ parent: [] });
     });
-    it('if adding clicked child means all children are selected, make selection array empty', () => {
-      const payload = getNewMapAfterChildAgentClick(
-        'agent4',
-        'state3',
-        ['state1', 'state2', 'state3'],
-        mockVisibilitySelectionMap
-      );
-      expect(payload).toEqual({
-        agent1: ['agent1'],
-        agent2: ['agent2'],
-        agent3: [],
-        agent4: [],
-        agent5: ['state1'],
+    it('adds child to parents selection array if child was not previously included', () => {
+      const result = getSelectionAfterChildCheckboxClick('child', 'parent', {
+        parent: ['child2'],
       });
+      expect(result).toEqual({ parent: ['child2', 'child'] });
     });
-    it('if adding clicked child is leaves at least one child unselected, add clicked state to array', () => {
-      const payload = getNewMapAfterChildAgentClick(
-        'agent5',
-        'state2',
-        ['state1', 'state2', 'state3'],
-        mockVisibilitySelectionMap
-      );
-      expect(payload).toEqual({
-        agent1: ['agent1'],
-        agent2: ['agent2'],
-        agent3: [],
-        agent4: ['state1', 'state2'],
-        agent5: ['state1', 'state2'],
-      });
+    it('will throw an error if the parent is not found in the userChangesMap', () => {
+      expect(() => {
+        getSelectionAfterChildCheckboxClick('child', 'parent', {});
+      }).toThrow('Parent not found in map');
     });
   });
 
+  describe('getChildren', () => {
+    it('returns the display states for an agent', () => {
+      const agent = {
+        name: 'name',
+        displayStates: [{ name: 'state1', id: '1', color: '#000000' }],
+        color: '#000000',
+      };
+      const result = getChildren(agent);
+      expect(result).toEqual(['1']);
+    });
+    it('returns an empty array if the agent has no display states', () => {
+      const agent = {
+        name: 'name',
+        displayStates: [],
+        color: '#000000',
+      };
+      const result = getChildren(agent);
+      expect(result).toEqual([]);
+    });
+  });
 
   describe('convertMapToSelectionStateInfo', () => {
-    it('returns an array of SelectionEntry objects', () => {
-      const payload = convertMapToSelectionStateInfo(
-        mockVisibilitySelectionMap
+    it('should return a selection entry with an empty tags array if agent name is in user changes map has agent name in array', () => {
```

Comment:
```suggestion
    it('should return a selection entry with an empty tags array if agent name is in user changes map', () => {
```

Reply:
will fix the typo (adding "and") but actually need that last phrase for accuracy